src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,173 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+
+    private int currentAmount = 0;
+    private transient Node<T> first;
+    private transient Node<T> last;
+
+    public void check(int index) {
+        if (index < 0 || index >= currentAmount) {
+            throw new ArrayIndexOutOfBoundsException("Non-valid index");
+        }
+    }
+
     @Override
     public boolean add(T value) {
-        return false;
+        int previousAmount = currentAmount;
+        addElement(value);
+        return currentAmount - previousAmount != 0;
```

Comment:
```suggestion
        return true;
```
if method completed its work correctly you can just return true, so this method in general looks redundant to me